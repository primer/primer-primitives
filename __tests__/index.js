expect.extend({
  toBeNonEmptyArray(value) {
    return {
      pass: Array.isArray(value) && value.length > 0,
      message: () => {
        return Array.isArray(value)
          ? `Expected .length > 0, but got ${value.length}`
          : `Expected an array, but got "${typeof value}"`
      }
    }
  },

  toBeSorted(value, comparator = numericAscending) {
    const sorted = [...value].sort(comparator)
    return {
      pass: this.equals(value, sorted),
      message: () => `Expected ${JSON.stringify(sorted)}, but got: ${JSON.stringify(value)}`
    }
  }
})

describe('theme (default export)', () => {
  const theme = require('..')

  it('is an object', () => {
    expect(theme).toBeInstanceOf(Object)
  })

  it('has a colors object', () => {
    expect(theme.colors).toBeInstanceOf(Object)
  })

  it('has a non-empty space array', () => {
    expect(theme.space).toBeNonEmptyArray()
  })

  it('has a fontSizes array', () => {
    expect(theme.fontSizes).toBeNonEmptyArray()
  })

  it('has a lineHeights object', () => {
    expect(theme.lineHeights).toBeInstanceOf(Object)
  })

  it('has a non-empty breakpoints array', () => {
    expect(theme.breakpoints).toBeNonEmptyArray()
  })

  it('has a maxWidths object', () => {
    expect(theme.maxWidths).toBeInstanceOf(Object)
  })

  it('matches the snapshot', () => {
    expect(theme).toMatchSnapshot()
  })
})

describe('colors export', () => {
  const colors = require('../colors')
  const expectedHues = ['gray', 'blue', 'green', 'yellow', 'orange', 'red', 'purple']

  it('is an object', () => {
    expect(colors).toBeInstanceOf(Object)
  })

  it('has a "black" string value', () => {
    expect(typeof colors.black).toBe('string')
  })

  it('has a "white" string value', () => {
    expect(typeof colors.white).toBe('string')
  })

  for (const hue of expectedHues) {
    it(`has a "${hue}" gradient array`, () => {
      expect(colors[hue]).toBeNonEmptyArray()
    })
  }

  it('matches the snapshot', () => {
    expect(colors).toMatchSnapshot()
  })
})

describe('space export', () => {
  const space = require('../space')

  it('is a non-empty array', () => {
    expect(space).toBeNonEmptyArray()
  })

  it('contains only numbers', () => {
    for (const n of space) {
      expect(typeof n).toBe('number')
    }
  })

  it('is sorted numerically', () => {
    expect(space).toBeSorted()
  })

  it('matches the snapshot', () => {
    expect(space).toMatchSnapshot()
  })
})

describe('typography export', () => {
  const typography = require('../typography')

  it('is an object', () => {
    expect(typography).toBeInstanceOf(Object)
  })

  describe('.fontSizes', () => {
    const {fontSizes} = typography
    it('is a non-empty array', () => {
      expect(fontSizes).toBeNonEmptyArray()
    })

    it('contains only numbers', () => {
      for (const n of fontSizes) {
        expect(typeof n).toBe('number')
      }
    })

    it('is sorted numerically', () => {
      expect(fontSizes).toBeSorted()
    })
  })

  describe('.lineHeights', () => {
    const {lineHeights} = typography
    it('is an object', () => {
      expect(lineHeights).toBeInstanceOf(Object)
    })

    it('matches the snapshot', () => {
      expect(lineHeights).toMatchSnapshot()
    })

    it('has a "default" key of 1.5', () => {
      expect(lineHeights).toMatchObject({default: 1.5})
    })
  })

  it('matches the snapshot', () => {
    expect(typography).toMatchSnapshot()
  })
})

function numericAscending(a, b) {
  return a - b
}
