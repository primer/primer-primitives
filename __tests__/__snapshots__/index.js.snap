// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colors export matches the snapshot 1`] = `
Object {
  "black": "#1b1f23",
  "blue": Array [
    "#f1f8ff",
    "#dbedff",
    "#c8e1ff",
    "#79b8ff",
    "#2188ff",
    "#0366d6",
    "#005cc5",
    "#044289",
    "#032f62",
    "#05264c",
  ],
  "gray": Array [
    "#fafbfc",
    "#f6f8fa",
    "#e1e4e8",
    "#d1d5da",
    "#959da5",
    "#6a737d",
    "#586069",
    "#444d56",
    "#2f363d",
    "#24292e",
  ],
  "green": Array [
    "#f0fff4",
    "#dcffe4",
    "#bef5cb",
    "#85e89d",
    "#34d058",
    "#28a745",
    "#22863a",
    "#176f2c",
    "#165c26",
    "#144620",
  ],
  "orange": Array [
    "#fff8f2",
    "#ffebda",
    "#ffd1ac",
    "#ffab70",
    "#fb8532",
    "#f66a0a",
    "#e36209",
    "#d15704",
    "#c24e00",
    "#a04100",
  ],
  "purple": Array [
    "#f5f0ff",
    "#e6dcfd",
    "#d1bcf9",
    "#b392f0",
    "#8a63d2",
    "#6f42c1",
    "#5a32a3",
    "#4c2889",
    "#3a1d6e",
    "#29134e",
  ],
  "red": Array [
    "#ffeef0",
    "#ffdce0",
    "#fdaeb7",
    "#f97583",
    "#ea4a5a",
    "#d73a49",
    "#cb2431",
    "#b31d28",
    "#9e1c23",
    "#86181d",
  ],
  "white": "#fff",
  "yellow": Array [
    "#fffdef",
    "#fffbdd",
    "#fff5b1",
    "#ffea7f",
    "#ffdf5d",
    "#ffd33d",
    "#f9c513",
    "#dbab09",
    "#b08800",
    "#735c0f",
  ],
}
`;

exports[`space export matches the snapshot 1`] = `
Array [
  0,
  4,
  8,
  16,
  24,
  32,
  40,
  48,
]
`;

exports[`theme (default export) matches the snapshot 1`] = `
Object {
  "breakpoints": Array [
    "1012px",
    "1280px",
    "544px",
    "768px",
  ],
  "colors": Object {
    "black": "#1b1f23",
    "blue": Array [
      "#f1f8ff",
      "#dbedff",
      "#c8e1ff",
      "#79b8ff",
      "#2188ff",
      "#0366d6",
      "#005cc5",
      "#044289",
      "#032f62",
      "#05264c",
    ],
    "gray": Array [
      "#fafbfc",
      "#f6f8fa",
      "#e1e4e8",
      "#d1d5da",
      "#959da5",
      "#6a737d",
      "#586069",
      "#444d56",
      "#2f363d",
      "#24292e",
    ],
    "green": Array [
      "#f0fff4",
      "#dcffe4",
      "#bef5cb",
      "#85e89d",
      "#34d058",
      "#28a745",
      "#22863a",
      "#176f2c",
      "#165c26",
      "#144620",
    ],
    "orange": Array [
      "#fff8f2",
      "#ffebda",
      "#ffd1ac",
      "#ffab70",
      "#fb8532",
      "#f66a0a",
      "#e36209",
      "#d15704",
      "#c24e00",
      "#a04100",
    ],
    "purple": Array [
      "#f5f0ff",
      "#e6dcfd",
      "#d1bcf9",
      "#b392f0",
      "#8a63d2",
      "#6f42c1",
      "#5a32a3",
      "#4c2889",
      "#3a1d6e",
      "#29134e",
    ],
    "red": Array [
      "#ffeef0",
      "#ffdce0",
      "#fdaeb7",
      "#f97583",
      "#ea4a5a",
      "#d73a49",
      "#cb2431",
      "#b31d28",
      "#9e1c23",
      "#86181d",
    ],
    "white": "#fff",
    "yellow": Array [
      "#fffdef",
      "#fffbdd",
      "#fff5b1",
      "#ffea7f",
      "#ffdf5d",
      "#ffd33d",
      "#f9c513",
      "#dbab09",
      "#b08800",
      "#735c0f",
    ],
  },
  "fontSizes": Array [
    12,
    14,
    16,
    20,
    24,
    32,
    40,
    48,
  ],
  "lineHeights": Object {
    "condensed": 1.25,
    "condensedUltra": 1,
    "default": 1.5,
  },
  "maxWidths": Object {
    "large": "1012px",
    "medium": "768px",
    "small": "544px",
    "xlarge": "1280px",
  },
  "space": Array [
    0,
    4,
    8,
    16,
    24,
    32,
    40,
    48,
  ],
}
`;

exports[`typography export .lineHeights matches the snapshot 1`] = `
Object {
  "condensed": 1.25,
  "condensedUltra": 1,
  "default": 1.5,
}
`;

exports[`typography export matches the snapshot 1`] = `
Object {
  "fontSizes": Array [
    12,
    14,
    16,
    20,
    24,
    32,
    40,
    48,
  ],
  "lineHeights": Object {
    "condensed": 1.25,
    "condensedUltra": 1,
    "default": 1.5,
  },
}
`;
